### PR TITLE
Detect target fix and competition logging

### PR DESCRIPTION
**Pull request by Xierumeng**
Arguments for Ultralytics prediction.

### CODE REVIEW

File: modules/detect_target/detect_target.py [hunk 1]
```diff
@@ -15,23 +16,35 @@ class DetectTarget:
...
         if len(predictions) == 0:
             return False, None
 
         # TODO: Change this to image points for image and telemetry merge for 2024
         # (bounding box conversion code required)
-        image_annotated = predictions[0].plot(show_conf=True)
+        image_annotated = predictions[0].plot(conf=True)
+
+        # Logging
+        if self.__filename_prefix != "":
```
Comment (mgupta27) [inline modules/detect_target/detect_target.py:45]: Can we also log the bounding box to a txt file with the same name as the corresponding image. Reason being @YMRTZ mentioned that he needed a data for the VMGM.

**Review by mgupta27 — COMMENTED** [review]
Overall looks good. Just a nit and a request from me.

File: modules/detect_target/detect_target.py [hunk 2]
```diff
@@ -15,23 +16,35 @@ class DetectTarget:
...
         Returns annotated image
         TODO: Change to PointsAndTime
         """
         image = data.frame
-        predictions = self.model.predict(image, stream=False)
+        predictions = self.__model.predict(
+            source=image,
+            half=True,
+            device=0,
+            stream=False
```
[the remark this answers is not in the document]
Comment (Xierumeng) [reply modules/detect_target/detect_target.py:34]: Sure.

File: modules/detect_target/detect_target.py (hunk 1 above)
Comment (Xierumeng) [reply modules/detect_target/detect_target.py:60]: Done.

**Review by mgupta27 — APPROVED** [review]
LGTM

**Pull request merged**